### PR TITLE
Heading level structure reset

### LINKED ISSUE #50: Heading level structure
Heading levels are really important for accessibility, as most screen reader users will start navigating a page by heading levels. There are a couple of key "rules" for heading levels:

- There should only be one `h1`
- Heading levels should only increase by one at a time (e.g. `h5` can't follow `h2`)

Using the axe chrome extension I can see we have a few errors flagged for heading level issues:

- The `h1` used for the Codeland logo is empty (we can fix this by adding alt text to the image)
- Heading levels increase by more than one at a time in various places

I had a quick inspection of just the home page headings, and I think we have an additional issue, in that headings are used for content that isn't a heading. For example the "hero section description" shouldn't really be a heading, but just large text:

screenshot of the heading below the codeland conf logo that describes the conference

It would be great for accessibility to review the headings we're using, and separate the visual text size from the heading level, controlling this with CSS instead. This will help make sure screen reader users can navigate our site easily.

### PR DESCRIPTION
**Pull request by danieltott**
- closes #50 

### Changes:
- Added utility classes to heading styles (`.text-h2`, `.text-h3` etc)
- Updated heading levels throughout the app

### Notes:
There are a lot of other accessibility/aria/etc issues going on that I intentionally left alone, will be getting to some of those in #51

### CODE REVIEW

File: source/layouts/_overview.erb [hunk 1]
```diff
@@ -1,34 +1,33 @@
-<section id="overview" class="section-card overview">
+<section id="overview" class="section-card overview" aria-label="CodeLand 2021 Overview">
```
Comment (aitchiss) [inline source/layouts/_overview.erb:1]: Nice 🙌

File: source/layouts/_sponsors.erb [hunk 2]
```diff
@@ -13,8 +13,8 @@
       <!-- <link href="//cdn-images.mailchimp.com/embedcode/classic-10_7.css" rel="stylesheet" type="text/css"> -->
       <div id="mc_embed_signup">
         <form action="https://codenewbie.us8.list-manage.com/subscribe/post?u=0a92a565e5b5037722c845fdb&amp;id=7736126f21" method="post" id="mc-embedded-subscribe-form" name="mc-embedded-subscribe-form" class="validate" target="_blank" novalidate>
-          <div id="mc_embed_signup_scroll">
-            <h2>Contact us</h2>
+          <fieldset id="mc_embed_signup_scroll">
+            <legend class="text-h2">Contact us</legend>
```
Comment (aitchiss) [inline source/layouts/_sponsors.erb:17]: ❤️

**Review by aitchiss — APPROVED** [review]
Nice one! This all looks good to me ✨

**Review by nickytonline — APPROVED** [review]
Thanks for making the changes @danieltott!

**Pull request merged**